### PR TITLE
Fix relabelings

### PR DESCRIPTION
**Pull request by hervenicol**
* Changes the way we add relabelings to serviceMonitors and podMonitors. Before, it was overriding the whole endpoint setups, now it just adds required relabelings.
* upgrade kyverno version used for tests from 1.5 to 1.6
* improve documentation for local tests

Towards https://github.com/giantswarm/giantswarm/issues/22922

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by QuentinBisson — APPROVED** [review]

File: helm/kyverno-policies-observability/templates/PodMonitor.yaml [hunk 1]
```diff
@@ -7,47 +7,60 @@ spec:
...
+        foreach:
+        - list: request.object.spec.podMetricsEndpoints
+          patchesJson6902: |-
+            # Pod Monitors are currently only supported on management clusters, so we can enforce the cluster_id and cluster_type.
+            - path: "/spec/podMetricsEndpoints/{{ `{{` }}elementIndex{{ `}}` }}/relabelings/-1"
+              op: add
+              value: {"replacement": "{{ .Values.managementCluster.name }}", "targetLabel": "cluster_id"}
+            - path: "/spec/podMetricsEndpoints/{{ `{{` }}elementIndex{{ `}}` }}/relabelings/-1"
+              op: add
+              value: {"replacement": "management_cluster", "targetLabel": "cluster_type"}
```
Comment (TheoBrigitte) [inline helm/kyverno-policies-observability/templates/PodMonitor.yaml:24]: It is quite hard to read those files, any way we could improve this?
I see we are adding multiple values, is there a way to combine those into one element maybe?
Maybe add a clear comment on what is the result produced in the end.

File: Makefile.custom.mk [hunk 2]
```diff
@@ -35,8 +35,8 @@ tilt-up: ## Start Tilt
 # If you change kyverno version here remember to change it in the Tiltfile too
 .PHONY: install-kyverno
 install-kyverno:
-	kubectl create --context kind-$(KIND_CLUSTER_NAME) -f https://raw.githubusercontent.com/kyverno/kyverno/v1.5.1/definitions/release/install.yaml
-	kubectl wait --context kind-$(KIND_CLUSTER_NAME) --for=condition=ready pod -l app=kyverno -nkyverno
+	kubectl create --context kind-$(KIND_CLUSTER_NAME) -f https://raw.githubusercontent.com/kyverno/kyverno/release-1.6/config/release/install.yaml
+	kubectl wait --context kind-$(KIND_CLUSTER_NAME) --for=condition=ready --timeout=90s pod -l app=kyverno -nkyverno
```
Comment (TheoBrigitte) [inline Makefile.custom.mk:39]: Would be good to move the kyverno fix into a different PR.

File: README.md [hunk 3]
```diff
@@ -16,18 +16,20 @@ These scripts enable us to easily set up a local testing environment.
...
 
 Executing the integration tests can be done with this simple set of commands:
 ```bash
-make setup # Creates the kind cluster and installs all dependencies.
-dabs.sh --generate-metadata -c ./helm/policies-aws # Builds helm chart archive to be tested.
-./dats.sh --chart-file $(ls -1 -t policies-aws*.tgz | head -n 1) # Executes the tests related to the AWS policies against the kind cluster.
+make kind-create kind-get-kubeconfig install-kyverno # Creates the kind cluster and installs all dependencies.
+dabs.sh --generate-metadata -c ./helm/kyverno-policies-observability # Builds helm chart archive to be tested.
+cp build/kyverno-policies-observability-*.tgz kyverno-policies-observability.tgz
+./dats.sh --chart-file kyverno-policies-observability.tgz --app-tests-pytest-tests-dir helm/kyverno-policies-observability/tests/ats # Executes the tests
```
Comment (TheoBrigitte) [inline README.md:32]: Would be good to move this into a separate PR also.

File: helm/kyverno-policies-observability/templates/PodMonitor.yaml (hunk 1 above)
Comment (hervenicol) [reply helm/kyverno-policies-observability/templates/PodMonitor.yaml:24]: Actually these files (helm templates) are generated from `policies/observability/` files by `make generate`.
The `policies/observability/` files are a bit easier to read because there's no escaping of `{{`, but yet not ideal.

I'll try one big json to be merged to the relabellings, may be easier to read.

File: README.md (hunk 3 above)
Comment (hervenicol) [reply README.md:32]: I agree that doc fixes and test updates should be in one or two separate PRs. Let me work on that.

**Comment by hervenicol** [comment]
On top of the code review comments, I tested these on Gauss:
* `opsctl deploy --use-kubeconfig  kyverno-policies-observability@fix_relabelings` deploys the clusterrules, they are valid and accepted
* `opsctl deploy --use-kubeconfig  grafana@master` fails, there's an error when applying the rules.

So I have to fix that, do some tests with the existing podmonitors and testmonitors, and maybe improve the CI tests.

**Comment by hervenicol** [comment]
Did not manage to reproduce the error.
Deleting the serviceMonitor fixed it, it was created again correctly. :shrug:

File: policies/observability/PodMonitor.yaml [hunk 4]
```diff
@@ -6,47 +6,79 @@ spec:
...
+                  "targetLabel": "cluster_id"
+                },
+                {
+                  "replacement": "management_cluster",
+                  "targetLabel": "cluster_type"
+                },
+                # Since we only support management clusters, all resources are considered as highest service priority
+                {
+                  "replacement": "highest",
+                  "targetLabel": "service_priority"
```
Comment (QuentinBisson) [inline policies/observability/PodMonitor.yaml:31]: It's not a big change but can you set the priority to highest only if the value is not set before ? Using the regex: nil or something similar ?

File: helm/kyverno-policies-observability/templates/PodMonitor.yaml (hunk 1 above)
Comment (hervenicol) [reply helm/kyverno-policies-observability/templates/PodMonitor.yaml:24]: Didn't manage to have the exact same behaviour with a big json. I could replace all relabelings, but not append relabelings.
Asked Kyverno people for help, they did not have a better idea that the current syntax. :shrug:

File: policies/observability/PodMonitor.yaml (hunk 4 above)
Comment (hervenicol) [reply policies/observability/PodMonitor.yaml]: It's a separate change IMO.
* could change some service and pod's `service_priority`
* testing is more complex, it requires a metrics source with specific labels

So it's a bit time consuming, I'd rather do it later / in a separate PR.

**Review by TheoBrigitte — APPROVED** [review]
LGTM
